standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -5390,6 +5406,493 @@ public void countOpenTxns() throws MetaException {
+              numDelete = 0;
+            }
+            LOG.debug("Removed COLUMN_STATS_ACCURATE from the parameters of the partition "
+                    + colStats.getStatsDesc().getDbName() + "." + colStats.getStatsDesc().getTableName() + "."
+                    + colStats.getStatsDesc().getPartName());
+          } else {
+            statementUpdate.setString(1, newParameter.get(COLUMN_STATS_ACCURATE));
+            statementUpdate.setLong(2, partId);
+            statementUpdate.addBatch();
+            numUpdate++;
+            if (numUpdate == maxNumRows) {
+              statementUpdate.executeBatch();
+              numUpdate = 0;
+            }
+            LOG.debug(" Executing update " + statementUpdate);
+          }
+        }
+        result.put((String) entry.getKey(), newParameter);
+      }
+
+      if (numInsert != 0) {
+        statementInsert.executeBatch();
+      }
+
+      if (numUpdate != 0) {
+        statementUpdate.executeBatch();
+      }
+
+      if (numDelete != 0) {
+        statementDelete.executeBatch();
+      }
+
+      if (isAcidTable) {
+        LOG.info("ETL_PERF started updateWriteIdForPartitions ");
+        updateWriteIdForPartitions(dbConn, writeId, partIdList);
+        LOG.info("ETL_PERF done updateWriteIdForPartitions ");
+      }
+      return result;
+    } finally {
+      closeStmt(stmtInt);
+      closeStmt(statementInsert);
+      closeStmt(statementUpdate);
+      closeStmt(statementDelete);
+    }
+  }
+
+  private static class PartitionInfo {
+    long partitionId;
+    long writeId;
+    public PartitionInfo(long partitionId, long writeId) {
+      this.partitionId = partitionId;
+      this.writeId = writeId;
+    }
+  }
+
+  private Map<String, PartitionInfo> getPartitionInfo(Connection dbConn, String catName,
+                                                      String dbName, String tblName,
+                                                      List<String> partKeys) throws SQLException {
+    List<String> queries = new ArrayList<>();
+    StringBuilder prefix = new StringBuilder();
+    StringBuilder suffix = new StringBuilder();
+    PreparedStatement pStmt = null;
+    ResultSet rs = null;
+    Map<String, PartitionInfo> partitionInfoMap = new HashMap<>();
+
+    try {
+      long tblId;
+      Statement stmt = dbConn.createStatement();
+      rs = stmt.executeQuery("select \"TBL_ID\" from \"DBS\", \"TBLS\" where \"DBS\".\"NAME\" = "
+              + quoteString(dbName) + " and \"DBS\".\"CTLG_NAME\" = " + quoteString(catName)
+              + " and \"TBLS\".\"TBL_NAME\" = " + quoteString(tblName)
+              + " and \"DBS\".\"DB_ID\" = \"TBLS\".\"DB_ID\"");
+      if (rs.next()) {
+        tblId = rs.getLong(1);
+      } else {
+        throw new RuntimeException("Invalid table name" + catName + "." + dbName + "." + tblName);
+      }
+
+      prefix.append("select \"PARTITIONS\".\"PART_ID\", \"PARTITIONS\".\"WRITE_ID\", \"PARTITIONS\".\"PART_NAME\" "
```

Comment:
why are you using PARTITIONS ref for every column  when you are using single table in the select?

Reply:
done